### PR TITLE
Remove slash delimiter from NavigationMesh properties

### LINKED ISSUE #61781: [3.5 RC3 - Mono] NavigationMesh resource has unusual property names.
### Godot version

3.5 RC3 mono

### System information

Windows 10, Nvidia gtx 1660Ti, latest drivers, GLES3

### Issue description

Firstly, I'm not sure if this is a bug in some parser, intended or ... so I apologize if this is how it's meant to be.

If you check NavigationMesh you will find property names such (__ is a double underscore):
```
public NavigationMesh.SourceGeometryMode Geometry__sourceGeometryMode
public string Geometry__sourceGroupName
public float Cell__size
```

These come from some gdscript names which contain a slash (/) so I'm not sure what is an exact C# format for this that's why I'm not sure if this is a bug or not, I just can't find what else in godot to compare this to.

### Steps to reproduce

Check the NavigationMesh properties.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by smix8**
Remove slash delimiter from NavigationMesh properties.

Fixes #61781

### CODE REVIEW

**Review by fire — APPROVED** [review]
Summary: This solves the C# bug, and we're allowed to do backward compatibility changes in master.

**Comment by akien-mga** [comment]
See https://github.com/godotengine/godot/issues/61781#issuecomment-1150345153, provided that this works, I think it would be worth doing to preserve some more compatibility for the custom values defined by users (even if they need to be tweaked further). Otherwise as it stands they would outright lose the values when opening the scenes in 3.5 if cherry-picked.

**Comment by smix8** [comment]
I did some limited testing renaming the values to their old path in the .tscn file. When loading the resource the properties were converted correctly without breaking the resource, including the 2 more special renames

File: scene/resources/navigation_mesh.h [hunk 1]
```diff
@@ -61,6 +61,8 @@ class NavigationMesh : public Resource {
 protected:
 	static void _bind_methods();
 	virtual void _validate_property(PropertyInfo &property) const override;
+	bool _set(const StringName &p_name, const Variant &p_value);
```
Comment (akien-mga) [inline scene/resources/navigation_mesh.h:64]: Should also be in `#ifndef`.

**Review by akien-mga — APPROVED** [review]

File: doc/classes/NavigationMesh.xml [hunk 2]
```diff
@@ -73,75 +73,75 @@
...
-		<member name="edge/max_length" type="float" setter="set_edge_max_length" getter="get_edge_max_length" default="12.0">
+		<member name="edge_max_length" type="float" setter="set_edge_max_length" getter="get_edge_max_length" default="12.0">
 			The maximum allowed length for contour edges along the border of the mesh.
-			[b]Note:[/b] While baking, this value will be rounded up to the nearest multiple of [member cell/size].
+			[b]Note:[/b] While baking, this value will be rounded up to the nearest multiple of [member cell_size].
 		</member>
-		<member name="filter/filter_walkable_low_height_spans" type="bool" setter="set_filter_walkable_low_height_spans" getter="get_filter_walkable_low_height_spans" default="false">
-			If [code]true[/code], marks walkable spans as not walkable if the clearance above the span is less than [member agent/height].
+		<member name="filter_walkable_low_height_spans" type="bool" setter="set_filter_walkable_low_height_spans" getter="get_filter_walkable_low_height_spans" default="false">
+			If [code]true[/code], marks walkable spans as not walkable if the clearance above the span is less than [member agent_height].
```
Comment (akien-mga) [inline doc/classes/NavigationMesh.xml:111]: Should be moved further down to respect alphabetical order.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**